org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/resourceloader/ParallelResourceLoader.java

```diff
@@ -88,6 +91,37 @@ protected Resource loadResource(URI uri, ResourceSet localResourceSet, ResourceS
 						// (if we do try to load resources that are out of sync, we create a deadlock)
 						return null;
 					}
+				} else if (uri.isArchive()) {
+					String authority = uri.authority();
+					if (authority != null && authority.startsWith("platform:/resource/")) {
```

Comment:
any way to get `platform:/resource` replaced by constants or methods on URI?

Reply:
Please give me that constant if you have one

Reply:
I'd think this should be a recursive to be able to handle nesting. Also we do have similar code in the Storage2URIMapper 

```
if (result.isEmpty() && uri.isArchive()) {
  String authority = uri.authority();
  authority = authority.substring(0, authority.length() - 1);
  URI archiveURI = URI.createURI(authority);
  if (archiveURI.isFile() || archiveURI.isPlatformResource()) {
```

Should we harmonize both locations?

Reply:
The constants are protected in URI so not generally reachable.

Reply:
In my Tests recursive does not work at all in Xtext

Reply:
And I did not find any exampl,e how recursive uri would look like

Reply:
=> i would need help for that since no idea how to test

Reply:
`jar:jar:platform:/resource/project/path/to/file.jar!/within/jar/nested.jar!/my/dsl.file` should be such a URI for nested jars in jars.

Reply:
I am not sure about that. And have no idea on how to test that. So I think I leave it was won't fix and we keep the workaround in the project forever